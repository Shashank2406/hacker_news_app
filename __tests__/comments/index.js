import React from 'react';
import 'react-native';
import {NativeModules, StatusBar} from 'react-native';
import Comments from '../../src/screens/comments';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});
import {shallow} from 'enzyme';

jest.useFakeTimers();
NativeModules.StatusBarManager = {getHeight: jest.fn()};
StatusBar.setBarStyle = jest.fn();

const navigation = {
  navigate: jest.fn(),
  addListener: jest.fn(),
  goBack: jest.fn(),
  push: jest.fn(),
};

const props = {
  navigation: navigation,
  route: {
    params: {
      selectedItem: '30620789',
    },
  },
};

it('Navigation Test', () => {
  const commentsScreenInstance = shallow(<Comments {...props} />).instance();
  expect(
    commentsScreenInstance.props.navigation.navigate,
  ).toHaveBeenCalledTimes(0);
});

it('State Verify', async () => {
  const commentsScreenInstance = shallow(<Comments {...props} />).instance();
  expect(commentsScreenInstance.state).toEqual({
    data: props.route.params.selectedItem || null,
    commentsData: [],
  });
});

it('fetchCommentsData Verify', async () => {
  const commentsScreenInstance = shallow(<Comments {...props} />).instance();
  const fetchCommentsDataSpy = jest.spyOn(
    commentsScreenInstance,
    'fetchCommentsData',
  );
  commentsScreenInstance.fetchCommentsData();
  expect(fetchCommentsDataSpy).toHaveBeenCalledTimes(1);
});

it('renders correctly', () => {
  const tree = renderer.create(<Comments {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
