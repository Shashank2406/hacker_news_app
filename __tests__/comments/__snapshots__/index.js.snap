// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 10,
        }
      }
    >
      undefined...
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      undefined
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
        undefined
      </Text>
      <View
        style={
          Object {
            "borderLeftColor": "#909090",
            "borderLeftWidth": 1,
            "marginHorizontal": 5,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
          }
        }
      >
        NaN seconds ago
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
          }
        }
      >
        undefined points
      </Text>
      <View
        style={
          Object {
            "borderLeftColor": "#909090",
            "borderLeftWidth": 1,
            "marginHorizontal": 5,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
          }
        }
      >
        0 comments
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "justifyContent": "center",
        "marginHorizontal": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      Comments
    </Text>
    <View
      style={
        Object {
          "borderTopWidth": 1,
          "marginVertical": 10,
        }
      }
    />
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListEmptyComponent={[Function]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginBottom": 150,
        }
      }
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <ActivityIndicator
          color="#000"
          size="large"
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
